pkg/startup/startup.go

```diff
@@ -250,6 +250,19 @@ func configureAndCheckIPAddressSubnets(ctx context.Context, cli client.Interface
 		terminate()
 	}
 
+	if node.Spec.BGP.IPv4Address == "" && node.Spec.BGP.IPv6Address == "" {
+		if os.Getenv("CALICO_NETWORKING_BACKEND") != "none" {
+			log.Error("No IP Addresses configured, and autodetection is not enabled")
+			// Unrecoverable if Calico is handling networking, terminate to restart.
+			terminate()
+		} else {
+			log.Warn("No IP Addresses configured, and autodetection is not enabled. Some features may not work properly.")
+			// Bail here setting node.Spec.BGP to nil so that the node validation passes.
+			node.Spec.BGP = nil
```

Comment:
I don't think this is the right action. The node BGP spec has a bunch of other fields in it that we don't just want to overwrite:

https://github.com/projectcalico/libcalico-go/blob/master/lib/apis/v3/node.go#L78-L93

I think we need to compare the BGP spec to an "empty" one, and if it is empty, then set it to nil.

Reply:
I wasn't able to think why this field would be populated for non-Calico networking cases. But, if we have to support what you are saying, I can do the empty comparison.

Reply:
I have added the BPGSpec comparison before settting it to `nil`